_source/_posts/2022-07-06-spa-web-security.md

```diff
@@ -0,0 +1,137 @@
+
+What's the takeaway here? Specifically, you'll want to make sure you have functioning certificates, ensure you're using the latest version of TLS, and make sure you use SSL/TLS over your _entire stack_. All of these steps might already be covered by your Cloud hosting provider. You should also verify you've added the **HTTP Strict Transport Security (HSTS)** security header to the web server, which adds an extra layer of security beyond HTTPS redirection.
+
+### Dive deeper into transport layer security 
+Check out these resources if you want to read more about Transport Layer Security and how to set up HTTPS redirection and HTST.
+* [Okta's API Security book chapter on Transport Layer Security](https://developer.okta.com/books/api-security/tls/)
+* [OWASP's Transport Layer Protection Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Transport_Layer_Protection_Cheat_Sheet.html)
+* [TLS Guidelines from MDN](https://infosec.mozilla.org/guidelines/web_security#transport-layer-security-tlsssl) 
+* [Getting started with HTTPS and certificates with Let's Encrypt](https://letsencrypt.org/getting-started/)
+
+## Keep your framework and external dependencies up to date
+You knew this was coming, even though we might lament the work involved to stay up to date. The advice to update your dependencies sounds like "Don't forget to floss," but it's essential (just like flossing). [Vulnerable and outdated components take the sixth spot in the OWASP Top Ten list](https://owasp.org/Top10/A06_2021-Vulnerable_and_Outdated_Components/). And there were some significant vulnerabilities from dependencies in the news recently. So you know deep down that burying your head in the sand is not a good action plan.
+
+It's important to note that we should update outdated components **and** ensure we don't accidentally depend on a vulnerable dependency. We should pin dependency versions in our `package.json` and then generate and enforce the use of a lock file. That means running `npm ci` instead of `npm i` to force the exact versions defined in the `package-lock.json`. If you use Yarn, you will use `yarn install --frozen-lockfile`. This way, updating versions is an intentional act.
+
+That's it. Just do it. It's important. Outdated components are a source of _liability_.
+
+### Dive deeper into keeping your components up to date
+Check out a step-by-step guide on dependency management for JavaScript developers by the OWASP team - [NPM Security best practices](https://cheatsheetseries.owasp.org/cheatsheets/NPM_Security_Cheat_Sheet.html).
+
+## Load resources securely
+A particular concern for SPAs are cross-origin resources, that is, resources your web app uses that are not hosted on the same origin as your web application. When all the resources the front-end needs are available through the same origin, we have confidence that the resources our secure. Hopefully, we're not pwning ourselves! At least not intentionally, right?!
```

Comment:
```suggestion
Cross-origin resources are a particular concern for SPAs. Cross-origin resources (CORs) refer to resources your web app uses that are not hosted on the same origin as your web application. When all the resources the front-end needs are available through the same origin, we have confidence that the resources are secure. Hopefully, we're not _pwning_ ourselves! At least not intentionally, right?!
```
clarifying via active voice. Typo fix.

Reply:
Hey Alisa, I'm still finding the syntax/grammar in this opening phrase awkward to read/understand  - How about you just revise that opening statement so the grammar works, and then you can use the semi-colon to pause, clarify, and continue on:  
A particular concern for SPAs are cross-origin resources --> Cross-origin resources are a particular concern for SPAs; this refers to resources that are not hosted on the same origin....